### DISCUSSION TITLE
[Share] add tag shortcut

### DISCUSSION THREAD

**Discussion by nilleo** [windingwind/zotero-actions-tags#139] [Action Scripts]
### Description

A script for quickly adding tags. By configuring the settings, you can use shortcut keys to add tags.
The script is based on the code of [Zutilo](https://github.com/wshanks/Zutilo).

### Event

None

### Operation

Script

### Data

```javascript
//add tag shortcut
// @author nilleo
// @link https://github.com/windingwind/zotero-actions-tags/discussions/139
// @usage Select an item in the library and press the assigned shortcut keys

const ZoteroPane = require("ZoteroPane")
// Select tag tab of item pane
var tabIndex = 2;
var tabbox = ZoteroPane.document.getElementById('zotero-view-tabbox');
tabbox.tabs.selectedIndex = tabIndex;

// Focus new tag entry textbox
let header = ZoteroPane.document.querySelector(".tags-box-header");
if (header === null) {
  // Zotero version < 5.0.78
  let tagPane = ZoteroPane.document.getElementById('zotero-editpane-tags');
  if (tagPane._tagColors === undefined) {
    window.setTimeout(function() { tagPane.new(); }, 200);
  } else {
    tagPane.newTag();
  }
} else {
  let addButton = header.querySelector("button");
  addButton.click();
}
```

**Comment by wwzhang** [comment]
Is that possible to use this for batch editing on multiple items?

**Reply by windingwind** [reply]
This action only activates the create tag input box in the item panel, which is only for the current item.

**Reply by ChinJCheung** [reply]
How to Add Tags to Multiple Items in a Batch?

**Reply by windingwind** [reply]
For pre-defined tags, see readme. it's a built-in feature of this plugin.

if you want to dynamically input tags and add them to the selected items, then you'll do it in a custom script.

**Reply by ChinJCheung** [reply]
yes, I want to be able to input tags dynamically and add them to the selected items. However, I am unsure how to implement t in a custom script. Is there a pre-existing script for this task, or can it be modified from the above script?

**Comment by ulahcherubim** [comment]
Not working with 7.0.0-beta.64+8f8c0d079 and 1.0.0-beta.34.
